diff --git a/etc/aether/services/containers/groups/01-intranet/02-gateway/services/promtail.nix b/etc/aether/services/containers/groups/01-intranet/02-gateway/services/promtail.nix
index 893c042..fab90dd 100644
--- a/etc/aether/services/containers/groups/01-intranet/02-gateway/services/promtail.nix
+++ b/etc/aether/services/containers/groups/01-intranet/02-gateway/services/promtail.nix
@@ -3,8 +3,8 @@
 in {
   users = {
     groups = {
-      promtail = {};
       nginx = {};
+      promtail = {};
     };
 
     users.promtail = {
diff --git a/etc/aether/services/containers/groups/02-internet/02-gateway/services/promtail.nix b/etc/aether/services/containers/groups/02-internet/02-gateway/services/promtail.nix
index efd18ea..905a204 100644
--- a/etc/aether/services/containers/groups/02-internet/02-gateway/services/promtail.nix
+++ b/etc/aether/services/containers/groups/02-internet/02-gateway/services/promtail.nix
@@ -1,12 +1,16 @@
 {container, ...}: let
   inherit (container) self intranet-monitoring;
 in {
-  users.groups.promtail = {};
-  users.groups.nginx = {};
-  users.users.promtail = {
-    isSystemUser = true;
-    group = "promtail";
-    extraGroups = ["nginx"];
+  users = {
+    groups = {
+      nginx = {};
+      promtail = {};
+    };
+    users.promtail = {
+      isSystemUser = true;
+      group = "promtail";
+      extraGroups = ["nginx"];
+    };
   };
 
   # PROMTAIL
